frontend/model/contracts/group.js

```diff
@@ -256,6 +334,46 @@ DefineContract({
         archiveProposal(state, data.proposalHash)
       }
     },
+    'gi.contracts/group/removeMember': {
+      validate: (data) => {
+        return objectOf({
+          member: string,
+          memberID: string,
+          groupID: string,
+          // In case it happens in a big group (by proposal)
+          // we need to validate the associated proposal.
+          proposalHash: optional(string),
+          proposalPayload: optional(objectOf({
+            secret: string // NOTE: simulate the OP_KEY_* stuff for now
+          }))
+        })
+      },
+      process (state, { data, meta }) {
+        const rootState = sbp('state/vuex/state')
+        if (data.member === rootState.loggedIn.username) {
+          // If this member is re-joining the group, ignore the rest
+          // so the member doesn't remove themself again.
+          if (sbp('okTurtles.data/get', 'JOINING_GROUP')) {
+            return false
+          }
+        }
```

Comment:
Two things:

1. Per the comment above, this code should be moved into the sideEffect, not here. It's OK for the process to delete the profile, because it should be re-added once we join the group. (Of course, verify that's true..)
2. I think the outer `if` shouldn't be needed. It should be sufficient to check if `'JOINING_GROUP'` is set

Reply:
1. we still the verification 'JOINING_GROUP', because when re-joining the group, the member isn't re-added to the list.
2. will remove.

Reply:
EDIT:
2. It cannot be removed.  When rejoining a group that removed another member in the past (besides us) we want to perform the whole removal action. If we remove the outer if, the action gets "ignored".

Reply:
> we still the verification 'JOINING_GROUP', because when re-joining the group, the member isn't re-added to the list.

I don't understand, what do you mean the member isn't re-added to the list? In what scenario?

> It cannot be removed. When rejoining a group that removed another member in the past (besides us) we want to perform the whole removal action. If we remove the outer if, the action gets "ignored".

Maybe you can explain this to me on a call, as I'm not understanding.

I know intuitively that it does not make sense to not run the process function and yet to run the sideEffect, which seems to be what you're suggesting.

Reply:
> I don't understand, what do you mean the member isn't re-added to the list? Why not? They should be added later on via the group/inviteAccept message.

I'm sorry, we were talking about different things. I thought you suggested to remove the `if(data.member === rootState.loggedIn.username)` from `process` and `sideEffect`. But you meant to remove it only from `process`.

Removed.